src/components/UserInfo.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { User } from '../types/User';
+
+interface Props {
+  user: User;
+}
+
+const UserInfo: React.FC<Props> = ({ user }) => (
+  <div className="details">
+    {user && (
+      <>
+        <span className="post-detail">{user.name}</span>
+        <span className="post-detail">{user.email}</span>
+      </>
+    )}
```

Comment:
В Props ты прописал user как обязательное свойство, и делаешь проверку, существует ли пользователь до того, как отображаешь компонент
Еще одна проверка в компоненте лишняя
```suggestion
    <span className="post-detail">{user.name}</span>
    <span className="post-detail">{user.email}</span>
```